trustme/__init__.py

```diff
@@ -194,28 +194,59 @@ class CA(object):
           your trust store to trust this CA.
 
     """
-    def __init__(self):
+    def __init__(self, parent_cert=None):
+        self.parent_cert = parent_cert
         self._private_key = rsa.generate_private_key(
             public_exponent=65537,
             key_size=_KEY_SIZE,
             backend=default_backend()
         )
 
         name = _name(u"Testing CA #" + random_text())
+        issuer = name
+        sign_key = self._private_key
+        if self.parent_cert is not None:
+            sign_key = parent_cert._private_key
+            issuer = parent_cert._certificate.subject
+
         self._certificate = (
-            _cert_builder_common(name, name, self._private_key.public_key())
+            _cert_builder_common(name, issuer, self._private_key.public_key())
             .add_extension(
                 x509.BasicConstraints(ca=True, path_length=9), critical=True,
             )
+            .add_extension(
+                x509.KeyUsage(
+                    digital_signature=False,
+                    content_commitment=False,
+                    key_encipherment=False,
+                    data_encipherment=False,
+                    key_agreement=False,
+                    key_cert_sign=True,
+                    crl_sign=False,
```

Comment:
Did a bit of reading on this, and found: "This extension MUST be present and MUST be marked critical.  Bit positions for keyCertSign and cRLSign MUST be set.  If the Private Key is used for signing OCSP responses, then the digitalSignature bit MUST be set." – https://cabforum.org/baseline-requirements-certificate-contents/

So I think we should have `crl_sign=True`?

Reply:
Thanks, done. (The openssl issue remains, I still need to look into it.)